stream-tests/src/test/java/org/apache/pekko/stream/javadsl/FlowTest.java

```diff
@@ -237,6 +238,22 @@ public void mustBeAbleToUseMapWithResource() {
     Assert.assertFalse(gate.get());
   }
 
+  @Test
+  public void mustBeAbleToUseMapWithAutoCloseableResource() {
+    final AtomicInteger closed = new AtomicInteger();
+    Source.from(Arrays.asList("1", "2", "3"))
+        .via(
+            Flow.of(String.class)
+                .mapWithResource(
+                    () -> (AutoCloseable) closed::incrementAndGet,
+                    (resource, elem) -> elem))
+                .runWith(TestSink.create(system), system)
+                .request(4)
+                .expectNext("1", "2", "3")
+                .expectComplete();
+    Assert.assertEquals(closed.get(), 3);
```

Comment:
Should assert the Autoclosable is been closed after the stream is completed.

Reply:
✅ updated